test/flows/student_finance_calculator_flow_test.rb

```diff
@@ -0,0 +1,362 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class StudentFinanceCalculatorTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup { testing_flow StudentFinanceCalculatorFlow }
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: when_does_your_course_start?" do
+    setup { testing_node :when_does_your_course_start? }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of what_type_of_student_are_you? for any response" do
+        assert_next_node :what_type_of_student_are_you?, for_response: "2020-2021"
+      end
+    end
+  end
+
+  context "question: what_type_of_student_are_you?," do
+    setup do
+      testing_node :what_type_of_student_are_you?
+      add_responses when_does_your_course_start?: "2020-2021"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of how_much_are_your_tuition_fees_per_year? for any response" do
+        assert_next_node :how_much_are_your_tuition_fees_per_year?, for_response: "uk-full-time"
+      end
+    end
+  end
+
+  context "UK full time students" do
```

Comment:
We shouldn't use contexts to organise nodes. The reason for this is that there's many many permutations of contexts that could occur and there's no way to apply it logically and you'll end up testing the same node logic multiple times. Life ends up simpler if you fully test a node within a single context for that node.

Some of this is explained in the ADR. This example should be useful: https://github.com/alphagov/smart-answers/blob/main/docs/arch/003-testing-flows.md#asserting-a-question-goes-to-a-next-node